### PR TITLE
Milestone 3 - Controllers specs

### PR DESCRIPTION
**Pull request by mukhammadrizooff**
# In this branch I implement:
- Add RSpec as a gem to my project.
- Create a Request spec file for all my controllers.
- Make sure that for each action I check:
    - If response status was correct.
    - If a correct template was rendered.
    - If the response body includes correct placeholder text.

### CODE REVIEW

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi Rizo, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that some essential aspects of the ReadMe are missing. Kindly ensure to improve upon these aspects of your ReadMe:

- [x] Add clear and detailed instructions on _how to run tests on this project_.


Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: spec/users_request_spec.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
+require 'rails_helper'
+
+RSpec.describe 'Users', type: :request do
```
Comment (arslanbisharat) [inline spec/users_request_spec.rb]: - [x] You should have `users_spec.rb` in `spec/requests/` directory with `type: request`.

File: spec/posts_request_spec.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
+require 'rails_helper'
+
+RSpec.describe 'Posts', type: :request do
```
Comment (arslanbisharat) [inline spec/posts_request_spec.rb]: - [x] You should have  `posts_spec.rb` in `spec/requests/` directory with `type: request`

**Review by arslanbisharat — CHANGES_REQUESTED** [review]
Hi @mukhammadrizooff, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
 - [x] You have added linters and they are working. 
 - [x] Pull Request is descriptive.
 - [x] Readme is also descriptive too. 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- x
- y
- z

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/requests/posts_request_spec.rb [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+      get '/users/1/posts/1'
+      expect(response).to render_template(:post_details)
+    end
+
+    it 'is a success' do
+      get '/users/1/posts/1'
+      expect(response).to have_http_status(:ok)
+    end
+  end
+end
```
Comment (ahmetbozaci) [inline spec/requests/posts_request_spec.rb:27]: ### OPTIONAL
_I wrote it as an optional but this is a requirement that was not mentioned in the last review. Please fix it._
- [x] Spec file should include tests that ensure the response body includes the correct placeholder text.
 Example : `expect(response.body).to include("Welcome to user index page")`

File: spec/requests/users_request_spec.rb [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+      get user_path(1)
+      expect(response).to render_template(:user_details)
+    end
+
+    it 'is a success' do
+      get user_path(1)
+      expect(response).to have_http_status(:ok)
+    end
+  end
+end
```
Comment (ahmetbozaci) [inline spec/requests/users_request_spec.rb:27]: ### OPTIONAL
_I wrote it as an optional but this is a requirement that was not mentioned in the last review. Please fix it._
- [x] Spec file should include tests that ensure the response body includes the correct placeholder text.
 Example : `expect(response.body).to include("Welcome to user index page")`

**Review by ahmetbozaci — APPROVED** [review]
# STATUS: APPROVED 🥳 🥳 🥳 



@mukhammadrizooff  ✋🏼 
#### Your project is great. 🎉 I approve of it. There is nothing else to say other than... it's time to merge it. You can submit this project on your dashboard.

___ 

### Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
___

### Cheers and Happy coding! 💻 👏

aproved

___

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification._
___

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form.](https://airtable.com/shrQAqnBwek5a0O0s)_

[GitHub](https://github.com/ahmetbozaci) | [LinkedIn](https://linkedin.com/in/ahmetbozaci) | [Twitter](https://twitter.com/ahmtbozaci)

_Reviewed by [Ahmet Bozacı ](https://github.com/ahmetbozaci)_

**Pull request merged**